### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Ideas for further development:

- I'd like to see some really simple test/example for this.
- This definitely needs a man page. The help generated from argparser is good, but I generally don't like programs without man pages.
- I think `bscalc --MiB 15698412` should work. I often need to "parse" some "random" number of bytes and adding `B` to some copy-pasted value is annoying :-)
- Support for sectors. (I can use `*/512`, but I'm too lazy.)

File: tools/bs_calc.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                print(("%0." + str(n_places) + "f" + " %s") % (value, args.unit))
+        else:
+            in_bytes = "%d B" % int(result)
+            print(in_bytes)
+
+            format_str = "%" + str(len(in_bytes) - n_places) + "." + str(n_places) + "f"
+            for b_unit in b_units[1:]:
+                converted = (format_str % result.convert_to(b_unit))
+                # don't print "0.00 CRAZY_BIG_UNIT"
+                if converted.strip() != "0." + n_places * "0":
```
Comment (vojtechtrefny) [inline tools/bs_calc.py]: This doesn't work with negative numbers

```bs_calc.py "- 10 MiB"
-10485760 B
-10240.00 KiB
   -10.00 MiB
    -0.01 GiB
    -0.00 TiB
    -0.00 PiB
    -0.00 EiB
```

**Comment by vojtechtrefny** [comment]
Found few "bugs" (mostly just not returning nice error messages):

Modulo with integer traceback:
```
$ python3 bs_calc.py "4 B % 2"
Traceback (most recent call last):
...
AttributeError: 'int' object has no attribute '_c_size'
```

Zero division traceback:
```
$ python3 bs_calc.py "4 B / 0"
Traceback (most recent call last):
...
bytesize.bytesize.ZeroDivisionError: Division by zero
```

Parsing exponents fails:
```
$ python3 bs_calc.py "1e5 MiB"
Error while parsing expression: Failed to parse size spec: 1e5 MiB
```
but it works with negative powers:
```
$ python3 bs_calc.py "1e-5 MiB"
10 B
0.01 KiB
```

**Comment by vpodzime** [comment]
> Support for sectors. (I can use `*/512`, but I'm too lazy.)

Sector size depends on the device.

File: configure.ac [hunk 2]
```diff
@@ -99,4 +99,29 @@ AS_IF([test "x$with_tools" != "xno" && test "x$with_python3" != "xyes"],
       [])
 
 AC_OUTPUT
+
+dnl ==========================================================================
+echo "
+               libblockdev $VERSION
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply configure.ac]: Ooops :smile:

**Comment by vpodzime** [comment]
This needs some more work on adding the man page.

**Comment by vpodzime** [comment]
> This needs some more work on adding the man page.

DONE.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Pull request merged**